2-onchain-interactions.md

```diff
@@ -0,0 +1,284 @@
+# IRC-2 Onchain Interactions
+
+NOTE: Debatably should not be included in an IRC, this service implementation should live within the `@statechannels` repository. The main concern is other services need to be able to also access the data stored here, primarily the data around onchain transactions
+
+## Overview
+
+Throughout the lifecycle of a channel there two major types of onchain interactions required:
+
+- monitoring + storing contract events
+- sending transactions to chain (during adjudication, funding, withdrawing)
+
+Both of these functions are handled by an onchain service, which the channel wallet has access to.
```

Comment:
> which the channel wallet has acces to

Should the onchain tx service be an instance that we create externally & pass into the wallet, or a lump of code that the wallet imports & instantiates internally?

Reply:
I don't think the wallet needs to know about this service. It can be entirely external.

Reply:
Ahh gotcha so this line should probably be reworded a tad:
```suggestion
Both of these functions are handled by an onchain service which has access to the channel wallet.
```